src/react-components/home/HomePage.js

```diff
@@ -0,0 +1,100 @@
+import React, { useContext, useEffect } from "react";
+import { FormattedMessage, addLocaleData } from "react-intl";
+import en from "react-intl/locale-data/en";
+import classNames from "classnames";
+import configs from "../../utils/configs";
+import IfFeature from "../if-feature";
+import { Page } from "../layout/Page";
+import MediaTiles from "../media-tiles";
+import { CreateRoomButton } from "../input/CreateRoomButton";
+import { PWAButton } from "../input/PWAButton";
+import { useFeaturedRooms } from "./useFeaturedRooms";
+import styles from "./HomePage.scss";
+import mediaBrowserStyles from "../../assets/stylesheets/media-browser.scss";
+import discordLogoSmall from "../../assets/images/discord-logo-small.png";
+import { AuthContext } from "../auth/AuthContext";
+import { createAndRedirectToNewHub } from "../../utils/phoenix-utils";
+
+addLocaleData([...en]);
+
+export function HomePage() {
+  const featuredRooms = useFeaturedRooms();
+  const auth = useContext(AuthContext);
+
+  useEffect(() => {
+    const qs = new URLSearchParams(location.search);
+
+    // Support legacy sign in urls.
```

Comment:
We need to make sure this is still handled when using a "default room", since it wont actually hit this page. Also thinking ahead a bit, we are probably going to need to move this up a level since we don't want "plugin" versions of the page to have to remember to support the sign in flow (that was why we wanted to split these off into their own pages in the first place)

Reply:
There's currently a hook that will support this sign in flow in the HomePage API PR, but yes I would like to get rid of that so I'll move it into index.js now. My current thoughts are that we will deprecate the old sign-in urls, ship this to all Hubs Cloud instances, ask users to update their clients, and update the email template to use the new urls. At that point we can safely remove the redirect from the index.js page.

How does the default room get handled now? I haven't tested that yet.